xapian-letor/tests/api_letor.cc

```diff
@@ -291,37 +321,34 @@ DEFINE_TESTCASE(preparetrainingfile, generated)
     string file1;
     string file2;
     while (getline(if1, file1)) {
-	TEST(getline(if2, file2));
-	istringstream iss1(file1);
-	istringstream iss2(file2);
-	string temp1;
-	string temp2;
-	int i = 0;
-	while ((iss1 >> temp1) && (iss2 >> temp2)) {
-	    if (i == 0 || i == 1 || i == 21) {
-		TEST_EQUAL(temp1, temp2);
-	    } else {
-		size_t t1 = temp1.find_first_of(':');
-		size_t t2 = temp2.find_first_of(':');
-		TEST(abs(stod(temp1.substr(t1 + 1)) -
-			 stod(temp2.substr(t2 + 1))) <
-		     0.001);
-	    }
-	    i++;
+    TEST(getline(if2, file2));
+    istringstream iss1(file1);
+    istringstream iss2(file2);
+    string temp1;
+    string temp2;
+    int i = 0;
+    while ((iss1 >> temp1) && (iss2 >> temp2)) {
+	if (i == 0 || i == 1 || i == 21) {
+	TEST_EQUAL(temp1, temp2);
+	} else {
+	size_t t1 = temp1.find_first_of(':');
+	size_t t2 = temp2.find_first_of(':');
+	TEST_EQUAL_DOUBLE(stod(temp1.substr(t1 + 1)),
+			  stod(temp2.substr(t2 + 1)));
```

Comment:
This block is now underindented (it looked OK before to me).  More cases of this below.

Reply:
That means that the code should look something like this? 
```
while ((line1 >> temp1) && (line2 >> temp2)) {
	if (i == 0 || i == 1 || i == 21) {
	TEST_EQUAL(temp1, temp2);
	} else {
	    size_t t1 = temp1.find_first_of(':');
	    size_t t2 = temp2.find_first_of(':');
	    TEST_EQUAL_DOUBLE(stod(temp1.substr(t1 + 1)),
			      stod(temp2.substr(t2 + 1)));
	}
	i++;
    }
```
instead of 
```
while ((line1 >> temp1) && (line2 >> temp2)) {
	if (i == 0 || i == 1 || i == 21) {
	    TEST_EQUAL(temp1, temp2);
	} else {
        size_t t1 = temp1.find_first_of(':');
	size_t t2 = temp2.find_first_of(':');
	TEST_EQUAL_DOUBLE(stod(temp1.substr(t1 + 1)),
			  stod(temp2.substr(t2 + 1)));
	}
	i++;
    }
```

Reply:
Neither of those is correct, but the second is incorrect in more ways - the first is just missing an indent of 4 columns on the first line.

Set up your editor so that a tab character in the file displays as moving to the next multiple of 8 column (so from the start of a line, a tab character moves from column 0 to 8, two tab characters to 16, etc).  It seems a tab display width of 4 is popular on Microsoft platforms, but 8 is the Unix convention and what we use.

Set the "shift indent" (or whatever your editor calls it) to 4 columns - each indent level (with a few exceptions) should be 4 columns.  So one indent from the start of a line is 4 spaces, two indent levels is a tab character, 3 is a tab character followed by 4 spaces, 4 is two tab characters, etc.

If your editor is too limited to handle the distinction (apparently some are), I'd strongly suggest setting it up for tab display width 8 and then managing the indenting by hand.  If you set tab display width to anything other than 8 you'll inevitably have a very confusing time trying to read existing code.

If you're using vim or neovim, then this should be very close to correct for Xapian code (the lines starting `"` are just comments in vim config file syntax):

```
set ts=8
set sw=4
set sts=4
" Fix indent after: case 0: {
" Indent public/protected/private by half sw
" Function return type gets no indent
" Line up in column after unclosed (
set cinoptions=l1,g0.5s,h0.5s,t0,(0
```

Reply:
@ojwb , would this be better? To include the while above? 
```
    while (getline(if1, line1)) {
	TEST(getline(if2, line2));
	istringstream iss1(line1);
	istringstream iss2(line2);
	string temp1;
	string temp2;
	int i = 0;
	while ((iss1 >> temp1) && (iss2 >> temp2)) {
	    if (i == 0 || i == 1 || i == 21) {
		TEST_EQUAL(temp1, temp2);
	    } else {
		size_t t1 = temp1.find_first_of(':');
		size_t t2 = temp2.find_first_of(':');
		TEST_EQUAL_DOUBLE(stod(temp1.substr(t1 + 1)),
				  stod(temp2.substr(t2 + 1)));
	    }
	    i++;
	}
	TEST_REL(i, ==, 22);
	TEST(!(iss2 >> temp2));
    }
    TEST(!getline(if2, line2));
```

Reply:
Yes.